bindings/go/descriptor/runtime/descriptor.go

```diff
@@ -0,0 +1,281 @@
+	// to indicate the resource content should not be part of the signature.
+	NoDigest = "NO-DIGEST"
+)
+
+// These constants describe identity attributes predefined by the
+// model used to identify elements (resources, sources and references)
+// in a component version.
+const (
+	IdentityAttributeName    = "name"
+	IdentityAttributeVersion = "version"
+)
+
+// Descriptor defines a schema specific descriptor of a component containing additionally embedded signatures
+// verifying the validity of the component.
+type Descriptor struct {
+	// Meta defines the schema version of the component.
+	Meta Meta `json:"-"`
+	// Component defines the component spec.
+	Component Component `json:"-"`
+	// Signatures contains optional signing information.
+	Signatures []Signature `json:"-"`
+}
+
+func (d *Descriptor) String() string {
+	base := d.Component.String()
+	if d.Meta.Version != "" {
+		base += fmt.Sprintf(" (schema version %s)", d.Meta.Version)
+	}
+	return base
+}
+
+// Component defines a named and versioned component containing dependencies such as sources, resources and
+// references pointing to further component versions.
+type Component struct {
+	ComponentMeta `json:",inline"`
+	// RepositoryContexts defines the previous repositories of the component.
+	// See https://github.com/open-component-model/ocm-spec/blob/main/doc/01-model/03-elements-sub.md#repository-contexts
+	RepositoryContexts []runtime.Unstructured `json:"-"`
+	// Provider describes the provider type of component in the origin's context.
+	Provider runtime.Identity `json:"-"`
+	// Resources defines all resources that are created by the component and by a third party.
+	Resources []Resource `json:"-"`
+	// Sources defines sources that produced the component.
+	Sources []Source `json:"-"`
+	// References component dependencies that can be resolved in the current context.
+	References []Reference `json:"-"`
+}
+
+// ResourceRelation describes whether the component is created by a third party or internally.
+type ResourceRelation string
+
+const (
+	// LocalRelation defines a internal relation
+	// which describes a internally maintained resource in the origin's context.
+	LocalRelation ResourceRelation = "local"
+	// ExternalRelation defines a external relation
+	// which describes a resource maintained by a third party vendor in the origin's context.
+	ExternalRelation ResourceRelation = "external"
+)
+
+// A Resource is a delivery artifact, intended for deployment into a runtime environment, or describing additional content,
+// relevant for a deployment mechanism.
+// For example, installation procedures or meta-model descriptions controlling orchestration and/or deployment mechanisms.
+// See https://github.com/open-component-model/ocm-spec/blob/main/doc/01-model/02-elements-toplevel.md#resources
+// +k8s:deepcopy-gen=true
+type Resource struct {
+	ElementMeta `json:",inline"`
+	// SourceRefs defines a list of source names.
```

Comment:
```suggestion
	// SourceRefs defines a list of sources used to generate the resource.
```
The "used to generate the resource" part is kind of optional. `source names` is just not really accurate for the `SourceRefs` data structure.